File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/dtos/UserResponse.java
package dev.jefy.connectpro.user.applicaion.dtos;


import java.util.UUID;

import dev.jefy.connectpro.shared.application.dtos.PortfolioSummaryData;
import dev.jefy.connectpro.user.domain.model.AuthUser;
import dev.jefy.connectpro.user.domain.vo.UserRole;

/**
 * @author Jôph Yamba
 */

public record UserResponse (
        UUID id,
        String email,
        String name,
        String imageUrl,
        UserRole role,
        PortfolioSummaryData portfolio
) {
    public static UserResponse fromUser(AuthUser user) {
        return new UserResponse(
                user.id(),
                user.email(), 
                user.name(),
                user.imageUrl(),
                user.role(),
                user.portfolio()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/user/application/dtos/UserResponse.kt
package dev.jefy.connectpro.user.application.dtos

import dev.jefy.connectpro.shared.application.dtos.PortfolioData
import dev.jefy.connectpro.user.domain.model.AuthUser
import dev.jefy.connectpro.user.domain.vo.UserRole
import java.util.*

data class UserResponse(
    val id: UUID,
    val email: String,
    val name: String?,
    val image: String?,
    val role: UserRole,
    val portfolio: PortfolioData?
)

fun AuthUser.toUserResponse(): UserResponse = UserResponse(
    id = this.id,
    email = this.email,
    name = this.fullname,
    image = this.image,
    role = this.role,
    portfolio = this.portfolio
)